File: src/test/java/com/ioByte/ByteChannelAdapter.java
package com.ioByte;

import com.socket.ActionData;
import com.socket.ActionHandler;
import com.socket.IoSession;

public class ByteChannelAdapter implements ActionHandler<Object> {


    @Override
    public void execute(ActionData<Object> actionData, IoSession session) {
        System.out.println(actionData.getAction());
        System.out.println(new String(actionData.getBuf()));
    }

}

File: src/test/kotlin/com/ioByte/ByteChannelAdapter.kt
package com.ioByte

import com.socket.ActionData
import com.socket.ActionHandler
import com.socket.IoSession

class ByteChannelAdapter : ActionHandler<Any> {

    override fun execute(actionData: ActionData<Any>, session: IoSession) {
        println("${actionData.action}: ${actionData.buf?.decodeToString()}")
    }
}
